### PR TITLE
Mermaid and Plant uml Parsers

### PR DESCRIPTION
**Pull request by Elektrum-77**
I made the mermaid and plant uml parsers that convert a Package into a String
I added a Main that prints a test package in the mermaid and plant uml format.

### CODE REVIEW

File: src/main/java/com/github/magickoders/Main.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  private static void crash(String errorMessage) {
+    System.err.println(errorMessage);
+    System.exit(1);
+  }
+
+  private static void writeToPath(Path path, String content) {
+    try (var writer = Files.newBufferedWriter(path)) {
+      writer.write(content);
+    } catch (IOException e) {
```
Comment (forax) [inline src/main/java/com/github/magickoders/Main.java:25]: you should propagate the IOException no try to recover from it here

File: src/main/java/com/github/magickoders/MermaidParser.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.github.forax.umldoc.core.Entity;
+import com.github.forax.umldoc.core.Field;
+import com.github.forax.umldoc.core.Package;
+import java.util.stream.Collectors;
+
+/**
+ * Converts a Package into as a mermaid formatted String.
+ */
+public record MermaidParser() implements Parser {
```
Comment (forax) [inline src/main/java/com/github/magickoders/MermaidParser.java:11]: why it's a record ??

File: src/main/java/com/github/magickoders/PlantumlParser.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.github.forax.umldoc.core.Entity;
+import com.github.forax.umldoc.core.Field;
+import com.github.forax.umldoc.core.Package;
+import java.util.stream.Collectors;
+
+/**
+ * Converts a Package into as a plantuml formatted String.
+ */
+public record PlantumlParser() implements Parser {
```
Comment (forax) [inline src/main/java/com/github/magickoders/PlantumlParser.java:11]: same here

**Comment by Elektrum-77** [comment]
The fixes are pushed.

**Comment by Elektrum-77** [comment]
The fixes are done

**Pull request merged**